build.sh

```diff
@@ -22,6 +22,34 @@ TAG="pihole:local"
 DOCKER_BUILD_CMD="docker buildx build src/. --tag ${TAG} --load --no-cache"
 FTL_FLAG=false
 
+# Check if buildx is installed
+docker buildx version >/dev/null 2>&1
+if [ $? -ne 0 ]; then
+    echo "Error: Docker buildx is required to build this image. For installation instructions, see:"
+    echo "       https://github.com/docker/buildx#installing"
+    exit 1
+fi
+
+# Function to check if a custom branch entered by the user is valid
+check_branch_exists() {
+    local repo=$1
+    local branch=$2
+    local url
+
+    if [ "$repo" == "ftl" ]; then
+        # Special case for FTL - we check for the binary instead of just the branch - in case it is not yet built.
+        url="https://ftl.pi-hole.net/${branch}/pihole-FTL-amd64"
+    else
+        url="https://github.com/pi-hole/${repo}/blob/${branch}/README.md"
```

Comment:
Maybe it doesn't matter, but do you think we should request the raw file instead of the Github page?

```suggestion
        url="https://raw.githubusercontent.com/pi-hole/${repo}/${branch}/README.md"
```

Reply:
Sorry, but this looks ugly. Please do it the git way

https://stackoverflow.com/a/54533010

Reply:
This is the build script for running on your host machine. The repositories may not be on that machine. Sure they may be on mine, but the average user isn't going to have all the repos cloned.

Cloning the repos just to do a pre-check before kicking off the build of the Dockerfile is unnessacery.


Doesn't look like we can do that, not without authenticating, or cloning the repos.. which we don't need to do.

![image](https://github.com/user-attachments/assets/751056b6-e2d3-4183-9699-6935e365e19c)

Why is checking the http return code of a request to a well-known ugly? I'm already using the same code to check the existence of the remote FTL binary, why not reuse that same code and check for the README file - which will always exist?

(If the check fails due to a lookup or internet issue, then the image isn't going to build anyway...)


We could just not do the check at all, and let the Docker build just crap out when it can't find the existing branch

https://github.com/pi-hole/docker-pi-hole/blob/development/src/Dockerfile#L48-L49

But aside from hardening the build script, this is about giving a friendly output to the user running it without them needing to trawl back through the docker build log to see what went wrong.

Reply:
core is a redirection, `pi-hole/pi-hole` should work fine, e.g.
```
git ls-remote https://github.com/pi-hole/pi-hole | grep "refs/heads" | cut -d'/' -f3- -
```
runs fine from my home directory (which is not a git repo)

Reply:
Oh piss, yeah sorry I wrote `core` instead of `pi-hole` like some kind of idiot

Reply:
Discussed on Mattermost